### ISSUE TITLE
[prefer-output-readonly] support output() function

### ISSUE THREAD

**Issue by reduckted** [angular-eslint/angular-eslint#1871]
**Description and reproduction of the issue**

The [`output()`](https://angular.dev/api/core/output) function can be used to create an output property, rather than using the `@Output()` decorator. The `prefer-output-readonly` should also check these properties.

```JSON
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": ["error"]
  }
}
```

```TS
// This raises a warning.
@Output()
public valueChange: EventEmitter<number> = new EventEmitter();

// This does not raise a warning.
public valueChange: OutputEmitterRef<number> = output();
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `18.0.1` |
| `@typescript-eslint/parser`        | `7.8.0` |
| `ESLint`                           | `8.57.0` |
| `node`                             | `22.2.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 17.3.7
Node: 20.9.0
Package Manager: npm 10.1.0
OS: win32 x64

Angular: 17.3.8
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1703.7
@angular-devkit/build-angular   17.3.7
@angular-devkit/core            17.3.7
@angular-devkit/schematics      17.3.7
@angular/cli                    17.3.7
@schematics/angular             17.3.7
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
